Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -65,6 +65,52 @@ RenderGrid::RenderGrid(Element& element, RenderStyle&& style)
 
 RenderGrid::~RenderGrid() = default;
 
+bool RenderGrid::isExtrinsicallySized() const
+{
+    auto& gridStyle = style();
+    auto allTracksAreExtrinsicallySized = [&] {
+        for (auto& column : gridStyle.gridTrackSizes(GridTrackSizingDirection::ForColumns)) {
+            if (column.isContentSized())
+                return false;
+        }
+
+        for (auto& row : gridStyle.gridTrackSizes(GridTrackSizingDirection::ForRows)) {
+            if (row.isContentSized())
+                return false;
+        }
+        return true;
+    };
+
+    // Since we currently only check if the grid's logical width is auto, it being
+    // extrinsically sized in this regard depends on the formatting context it
+    // participates in. For now we only check and allow if it participates in block
+    // layout since that is simple.
+    auto participatesInBlockLayout = [&] {
+        if (isInline())
+            return false;
+        auto* containingBlock = this->containingBlock();
+        return containingBlock && containingBlock->isBlockContainer();
```

Comment:
This is OK but the spec accurate check would be looking at the containing block (block container) and see if it has inline or block level boxes (`containingBlock->childrenInline()`)

Reply:
Good point!